src/cocotb/types/_range.py

```diff
@@ -165,6 +166,12 @@ def __repr__(self) -> str:
 
     index = cached_method(Sequence.index)
 
+    def __copy__(self) -> "Range":
+        return Range.from_range(self._range)
+
+    def __deepcopy__(self, memo: Dict[int, Any]) -> "Range":
+        return Range.from_range(copy.deepcopy(self._range, memo=memo))
```

Comment:
`range` object is immutable.
```suggestion
        return Range.from_range(self._range)
```